server-spi/src/main/java/org/keycloak/models/UserCredentialModel.java

```diff
@@ -126,6 +127,14 @@ public static UserCredentialModel generateSecret() {
         return new UserCredentialModel("", SECRET, SecretGenerator.getInstance().randomString());
     }
 
+    public static UserCredentialModel buildFromBackupAuthnCode(String backupAuthnCodeInput) {
+        return new UserCredentialModel("", RecoveryAuthnCodesCredentialModel.TYPE, backupAuthnCodeInput);
```

Comment:
you could call `buildFromBackupAuthnCode("", backupAuthnCodeInput)` here

Reply:
OK.